### ISSUE TITLE
Simple SSL not working for MarkLogic

### ISSUE THREAD

**Issue by iowusu** [BillFarber/kafka-marklogic-connector#5]
**Issue** 
When SSL is turn on MarkLogic appserver, the kafka-marklogic-connector fails to start. 

**Error** 
ERROR WorkerSinkTask{id=marklogic-sink-0} Task threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask:179)
com.marklogic.client.ForbiddenUserException: Local message: User is not allowed to read resource at internal/forestinfo. Server Message: Server (not a REST instance?) did not respond with an expected REST Error message.
	at com.marklogic.client.impl.OkHttpServices.checkStatus(OkHttpServices.java:4321)
	at com.marklogic.client.impl.OkHttpServices.getResource(OkHttpServices.java:3056)
	at com.marklogic.client.datamovement.impl.DataMovementServices.readForestConfig(DataMovementServices.java:51)
	at com.marklogic.client.datamovement.impl.DataMovementManagerImpl.readForestConfig(DataMovementManagerImpl.java:160)
	at com.marklogic.client.datamovement.impl.DataMovementManagerImpl.getForestConfig(DataMovementManagerImpl.java:155)
	at com.marklogic.client.datamovement.impl.DataMovementManagerImpl.newWriteBatcher(DataMovementManagerImpl.java:103)
	at com.marklogic.kafka.connect.sink.MarkLogicSinkTask.start(MarkLogicSinkTask.java:37)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.initializeAndStart(WorkerSinkTask.java:300)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:189)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:177)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:227)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[2019-10-17 11:22:28,920] ERROR WorkerSinkTask{id=marklogic-sink-0} Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:180)

**Recreate the Error** 
set up marklogic appServer with SSL turn on
set up ml.connection.simpleSsl=true  on config/marklogic-sink.properties
start  the kakfka-marklogic-connector

**Comment by rjrudin** [comment]
I recommend making a Java program that tries to create a WriteBatcher against the ML app server that requires SSL. The docs shows a simple example of this - http://docs.marklogic.com/guide/java/data-movement#id_80024 . 

You’ll need to adjust how the DatabaseClient is created so that it uses SSL. I recommend using the same code that the connector is using - https://github.com/BillFarber/kafka-marklogic-connector/blob/master/src/main/java/com/marklogic/kafka/connect/DefaultDatabaseClientCreator.java#L72 (you don't need to use a DatabaseClientConfig, just use the same SSLContext and hostname verifier).  

The issue may be with the default SSLContext that your JVM is constructing. But there's no way to know for sure unless you try to create a WriteBatcher outside the context of the Kafka connector.

**Comment by BillFarber** [comment]
I was able to use the connector to send messages to a MarkLogic Rest server using SSL. I had to do two things:
1) In the marklogic-sink.properties properties file, you must set "ml.connection.simpleSsl=true”.
2) You must also add the ML server’s certificate (or the signing CA for that cert) to your cacerts file in your JVM path. (Mine was in java/lib/security).
Keep in mind that this only does simple SSL. It does not verify the hostname. It also does not implement client certificates yet.

Based on my success last night and the error message, I agree with Rob that this is not related to the Kafka connector.
